### PR TITLE
Add fixes from publishing

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
Fixing some things that got in the way of publishing.

### CODE REVIEW

File: lib/c-api/build.rs [hunk 1]
```diff
@@ -417,15 +417,25 @@ fn exclude_items_from_wasm_c_api(builder: Builder) -> Builder {
...
+        .unwrap()
+        .to_string_lossy()
+        .to_string()
+        .starts_with("wasmer-c-api"));
     shared_object_dir.pop();
 
-    shared_object_dir.push("target");
+    assert_eq!(shared_object_dir.file_name(), Some(OsStr::new("build")));
+    shared_object_dir.pop();
+    shared_object_dir.pop();
```
Comment (nlewycky) [inline lib/c-api/build.rs:436]:
```suggestion
    shared_object_dir.pop();  // "debug" or "release"
```

**Review by nlewycky — APPROVED** [review]

File: lib/c-api/build.rs [hunk 2]
```diff
@@ -417,15 +417,31 @@ fn exclude_items_from_wasm_c_api(builder: Builder) -> Builder {
 fn build_inline_c_env_vars() {
     use std::ffi::OsStr;
 
-    let mut shared_object_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    // We start from `OUT_DIR` because `cargo publish` uses a different directory
+    // so traversing from `CARGO_MANIFEST_DIR` is less reliable.
+    let mut shared_object_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
```
Comment (Hywan) [inline lib/c-api/build.rs:422]: Good catch!

**Pull request merged**